scripts/jenkins/jenkins-job-trigger

```diff
@@ -1,96 +1,65 @@
-my $CRUMB;
-my $METHOD='POST';
-
-# read the api credentials from files
-eval `cat /etc/jenkinsapi.cred ./jenkinsapi.cred 2>/dev/null`;
-
-if ( ! $APIKEY ) {
-  print STDERR "You have not defined your APIKEY variable in the config.\n";
-  print STDERR "Login in jenkins and find your api key.";
-  print STDERR "Please add this api key to your jenkinsapi.cred file:\n";
-  print STDERR ' $APIKEY="987zyx654wvu"';
-}
-
-if ( ! $CRUMB ) {
-  print STDERR "You have not defined your CRUMB variable in the config.\n";
-  print STDERR "Please create your own crumb:\n";
-  print STDERR " curl --user <myusername>:<mypassword> 'https://ci.opensuse.org/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,\":\",//crumb)'";
-  print STDERR "And then add it to your jenkinsapi.cred file:\n";
-  print STDERR ' $CRUMB=".crumb:123abc456def"';
-}
-
-$METHOD ||= 'POST';
-$SCHEME ||= 'https';
-if ($JURL =~ /river\.suse\.de/)
-{
-  $SCHEME='http';
-}
-
-my $up='';
-$up="${USER}:${APIKEY}\@" if (defined $USER && $USER ne '' && defined $APIKEY && $APIKEY ne '');
-my $jurl="$SCHEME://${up}${JURL}";
-my $job=shift || die usage();
-
-my $jobtype='normal';
-my $jtype=shift || '';
-if ($jtype =~ /^-p$/) {
-  $jobtype='parameter';
-} elsif ($jtype =~ /^-m$/) {
-  $jobtype='matrix';
-}
-
-my $triggerurl=$jurl."/job/".$job;
-my @curlargs=('--silent', '--show-error');
-push @curlargs, ("-H", "Content-Length: 0");
-
-if ($jobtype eq 'normal') {
-  $triggerurl .= "/build";
-} elsif ($jobtype eq 'parameter') {
-  $triggerurl .= "/buildWithParameters?".join('&', @ARGV);
-} elsif ($jobtype eq 'matrix') {
-  $triggerurl .= "/".join(',', @ARGV)."/build";
-}
-
-$triggerurl =~ s/ /+/g;
-push @curlargs, ("-X", $METHOD);
-push @curlargs, ("-H", $CRUMB) if ($CRUMB);
-my @cmd=("curl", @curlargs, $triggerurl);
-
-print join(' ', @cmd) if ( $ENV{debug} );
-exec(@cmd) || die $!;
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+import sys
+import json
+
+try:
+    import jenkins
+except ImportError:
+    print("Error: Dependency missing. Please install 'python-jenkins'")
+    sys.exit(1)
+
+config_files = [ '/etc/jenkinsapi.conf', './jenkinsapi.conf' ]
+config = dict()
+config_loaded = False
+job_parameters = dict()
+
+for config_file in config_files:
+    if not os.path.exists(config_file):
+        continue
+    with open(config_file, 'r') as f:
+        config.update(json.load(f))
+        config_loaded = True
+
+if not config_loaded:
+    print('Error: No config file could be loaded. Please create either of: %s' %
+          ', '.join(config_files))
+    sys.exit(1)
+
+try:
```

Comment:
Does it really make sense to use argparse here?
There it no point in creating a usage via argparse as the parameters can be anything (really ANYthing), so we do not have a list of supported parameters. Its not possible and not wanted. And it should be backwards compatible.
So just taking all elements from argv seemed to be the most logical thing to do.